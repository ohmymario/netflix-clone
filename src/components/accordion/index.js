import React, { useState, createContext, useContext } from 'react';
import PropTypes from 'prop-types';
import {
  Container,
  Frame,
  Title,
  Item,
  Inner,
  Header,
  Body,
} from './styles/accordion';

const ToggleContext = createContext();

const Accordion = (props) => {
  const { children, ...restProps } = props;
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
};

Accordion.Title = function AccordionTitle(props) {
  const { children, ...restProps } = props;
  return <Title {...restProps}>{children}</Title>;
};

Accordion.Frame = function AccordionFrame(props) {
  const { children, ...restProps } = props;
  return <Frame {...restProps}>{children}</Frame>;
};

Accordion.Item = function AccordionItem(props) {
  const [toggleShow, setToggleShow] = useState(false);
  const { children, ...restProps } = props;
  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

Accordion.Header = function AccordionHeader(props) {
  const { children, ...restProps } = props;
  const { toggleShow, setToggleShow } = useContext(ToggleContext);
  return (
    <Header onClick={() => setToggleShow(!toggleShow)} {...restProps}>
      {children}
      {toggleShow ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Accordion.Body = function AccordionBody(props) {
  const { children, ...restProps } = props;
  const { toggleShow } = useContext(ToggleContext);
  return toggleShow ? <Body {...restProps}>{children}</Body> : null;
};

Accordion.propTypes = {
  children: PropTypes.node,
};

export default Accordion;
