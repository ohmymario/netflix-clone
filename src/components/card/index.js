import React, { useState, useContext, createContext } from 'react';
import PropTypes from 'prop-types';
import {
  Container,
  Content,
  Entities,
  Feature,
  FeatureTitle,
  FeatureText,
  FeatureClose,
  Group,
  Title,
  SubTitle,
  Text,
  Maturity,
  Meta,
  Item,
  Image,
} from './styles/card';

export const FeatureContext = createContext();

const Card = (props) => {
  const { children, ...restProps } = props;
  const [showFeature, setShowFeature] = useState(false);
  const [itemFeature, setItemFeature] = useState({});
  return (
    <FeatureContext.Provider
      value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}
    >
      <Container {...restProps}>{children}</Container>
    </FeatureContext.Provider>
  );
};

Card.Group = function CardGroup(props) {
  const { children, ...restProps } = props;
  return <Group {...restProps}>{children}</Group>;
};

Card.Title = function CardTitle(props) {
  const { children, ...restProps } = props;
  return <Title {...restProps}>{children}</Title>;
};

Card.SubTitle = function CardSubTitle(props) {
  const { children, ...restProps } = props;
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Card.Entities = function CardEntities(props) {
  const { children, ...restProps } = props;
  return <Entities {...restProps}>{children}</Entities>;
};

Card.Text = function CardText(props) {
  const { children, ...restProps } = props;
  return <Text {...restProps}>{children}</Text>;
};

Card.Meta = function CardMeta(props) {
  const { children, ...restProps } = props;
  return <Meta {...restProps}>{children}</Meta>;
};

Card.Item = function CardItem(props) {
  const { setShowFeature, setItemFeature } = useContext(FeatureContext);
  const { children, item, ...restProps } = props;
  return (
    <Item
      onClick={() => {
        setItemFeature(item);
        setShowFeature(true);
      }}
      {...restProps}
    >
      {children}
    </Item>
  );
};

Card.Feature = function CardFeature(props) {
  const { children, category, ...restProps } = props;
  const { showFeature, setShowFeature, itemFeature } = useContext(
    FeatureContext
  );

  return showFeature ? (
    <Feature
      src={`/images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`}
      {...restProps}
    >
      <Content>
        <FeatureTitle>{itemFeature.title}</FeatureTitle>
        <FeatureText>{itemFeature.description}</FeatureText>
        <FeatureClose onClick={() => setShowFeature(false)}>
          <img src="/images/icons/close.png" alt="close" />
        </FeatureClose>

        <Group margin="30px 0" flexDirection="row" alignItems="center">
          <Maturity rating={itemFeature.maturity}>
            {itemFeature.maturity < 12 ? 'PG' : itemFeature.maturity}
          </Maturity>
          <FeatureText fontWeight="bold">
            {itemFeature.genre.charAt(0).toUpperCase() +
              itemFeature.genre.slice(1)}
          </FeatureText>
        </Group>
        {children}
      </Content>
    </Feature>
  ) : null;
};

Card.Image = function CardImage(props) {
  const { src, ...restProps } = props;
  return <Image src={src} {...restProps} />;
};

Card.propTypes = {
  children: PropTypes.node,
};

Card.Image.propTypes = {
  src: PropTypes.string,
};

Card.Feature.propTypes = {
  category: PropTypes.string,
};

Card.Item.propTypes = {
  item: PropTypes.object,
};

export default Card;
