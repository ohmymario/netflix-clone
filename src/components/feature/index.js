import React from 'react';
import PropTypes from 'prop-types';
import { Container, Title, Subtitle } from './styles/feature';

const Feature = (props) => {
  const { children, ...restProps } = props;
  return <Container {...restProps}>{children}</Container>;
};

Feature.Title = function FeatureTitle(props) {
  const { children, ...restProps } = props;
  return <Title {...restProps}>{children}</Title>;
};

Feature.Subtitle = function FeatureSubtitle(props) {
  const { children, ...restProps } = props;
  return <Subtitle {...restProps}>{children}</Subtitle>;
};

Feature.propTypes = {
  children: PropTypes.node,
};

export default Feature;
