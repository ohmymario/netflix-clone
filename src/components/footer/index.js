import React from 'react';
import PropTypes from 'prop-types';
import {
  Container,
  Row,
  Column,
  Link,
  Title,
  Text,
  Break,
} from './styles/footer';

const Footer = (props) => {
  const { children, ...restProps } = props;
  return <Container {...restProps}>{children}</Container>;
};

Footer.Row = function FooterRow(props) {
  const { children, ...restProps } = props;
  return <Row {...restProps}>{children}</Row>;
};

Footer.Column = function FooterColumn(props) {
  const { children, ...restProps } = props;
  return <Column {...restProps}>{children}</Column>;
};

Footer.Link = function FooterLink(props) {
  const { children, href, ...restProps } = props;
  return (
    <Link href={href} {...restProps}>
      {children}
    </Link>
  );
};

Footer.Title = function FooterTitle(props) {
  const { children, ...restProps } = props;
  return <Title {...restProps}>{children}</Title>;
};

Footer.Text = function FooterText(props) {
  const { children, ...restProps } = props;
  return <Text {...restProps}>{children}</Text>;
};

Footer.Break = function FooterBreak(props) {
  const { children, ...restProps } = props;
  return <Break {...restProps}>{children}</Break>;
};

Footer.propTypes = {
  children: PropTypes.node,
};

Footer.Link.propTypes = {
  children: PropTypes.node,
  href: PropTypes.string.isRequired,
};

export default Footer;
