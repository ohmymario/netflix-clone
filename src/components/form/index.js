import React from 'react';
import PropTypes from 'prop-types';
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  StyledLink,
  Input,
  Submit,
} from './styles/form';

const Form = (props) => {
  const { children, ...restProps } = props;
  return <Container {...restProps}>{children}</Container>;
};

Form.Base = function FormBase(props) {
  const { children, ...restProps } = props;
  return <Base {...restProps}>{children}</Base>;
};

Form.Error = function FormError(props) {
  const { children, ...restProps } = props;
  return <Error {...restProps}>{children}</Error>;
};

Form.Title = function FormTitle(props) {
  const { children, ...restProps } = props;
  return <Title {...restProps}>{children}</Title>;
};

Form.Text = function FormText(props) {
  const { children, ...restProps } = props;
  return <Text {...restProps}>{children}</Text>;
};

Form.TextSmall = function FormTextSmall(props) {
  const { children, ...restProps } = props;
  return <TextSmall {...restProps}>{children}</TextSmall>;
};

Form.Link = function FormLink(props) {
  // react-router-dom Link
  const { children, to, ...restProps } = props;
  return (
    <StyledLink to={to} {...restProps}>
      {children}
    </StyledLink>
  );
};

Form.Input = function FormInput(props) {
  const { children, placeholder, type, value, ...restProps } = props;
  return (
    <Input placeholder={placeholder} type={type} value={value} {...restProps}>
      {children}
    </Input>
  );
};

Form.Submit = function FormSubmit(props) {
  const { children, disabled, type, ...restProps } = props;
  return (
    <Submit disabled={disabled} type={type} {...restProps}>
      {children}
    </Submit>
  );
};

Form.propTypes = {
  children: PropTypes.node,
};

Form.Link.propTypes = {
  to: PropTypes.string,
};

Form.Input.propTypes = {
  placeholder: PropTypes.string,
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};

Form.Submit.propTypes = {
  disabled: PropTypes.bool,
  type: PropTypes.string.isRequired,
};

export default Form;
