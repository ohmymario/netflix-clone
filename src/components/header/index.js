import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Background, Container, Logo, ButtonLink } from './styles/header';

const Header = (props) => {
  const { children, bg = true, ...restProps } = props;
  return bg ? <Background {...restProps}>{children}</Background> : children;
};

Header.Frame = function HeaderFrame(props) {
  const { children, ...restProps } = props;
  return <Container {...restProps}>{children}</Container>;
};

Header.ButtonLink = function HeaderButtonLink(props) {
  const { children, to, ...restProps } = props;
  return (
    <ButtonLink to={to} {...restProps}>
      {children}
    </ButtonLink>
  );
};

Header.Logo = function HeaderLogo(props) {
  const { children, to, alt, src, ...restProps } = props;
  return (
    <Link to={to}>
      <Logo src={src} alt={alt} {...restProps} />
    </Link>
  );
};

Header.propTypes = {
  children: PropTypes.node,
  bg: PropTypes.bool,
};

Header.ButtonLink.propTypes = {
  children: PropTypes.node,
  to: PropTypes.string,
};

Header.Logo.propTypes = {
  children: PropTypes.node,
  to: PropTypes.string,
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
};

export default Header;
