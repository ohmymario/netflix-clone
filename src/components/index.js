// More simple imports from components folder

export { default as Jumbotron } from './jumbotron';
export { default as Footer } from './footer';
export { default as Accordion } from './accordion';
export { default as optForm } from './optForm';
