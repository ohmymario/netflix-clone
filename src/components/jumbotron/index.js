import React from 'react';
import PropTypes from 'prop-types';
import {
  Item,
  Inner,
  Container,
  Pane,
  Title,
  Subtitle,
  Image,
} from './styles/jumbotron';

// Hero unit style content

const Jumbotron = (props) => {
  // Grabbing restProps for compound component
  const { children, direction = 'row', ...restProps } = props;
  return (
    <Item {...restProps}>
      <Inner direction={direction}>{children}</Inner>
    </Item>
  );
};

Jumbotron.Container = function JumbotronContainer(props) {
  const { children, ...restProps } = props;
  return <Container {...restProps}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane(props) {
  const { children, ...restProps } = props;
  return <Pane {...restProps}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle(props) {
  const { children, ...restProps } = props;
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.Subtitle = function JumbotronSubtitle(props) {
  const { children, ...restProps } = props;
  return <Subtitle {...restProps}>{children}</Subtitle>;
};

Jumbotron.Image = function JumbotronImage(props) {
  const { src, alt, ...restProps } = props;
  return <Image src={src} alt={alt} {...restProps} />;
};

Jumbotron.propTypes = {
  children: PropTypes.node,
  direction: PropTypes.string,
};

Jumbotron.Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
};

export default Jumbotron;
