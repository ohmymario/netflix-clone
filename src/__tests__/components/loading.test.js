import React from 'react';
import { render } from '@testing-library/react';
import { Loading } from '../../components';

describe('<Loading/>', () => {
  it('renders the <Loading/> component', () => {
    const { container, getByTestId } = render(
      <Loading src="" data-testid="loading" />
    );
    expect(getByTestId('loading')).toBeTruthy();
    expect(getByTestId('loading-picture')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Loading.ReleaseBody/>', () => {
    const { container, getByTestId } = render(<Loading.ReleaseBody />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
