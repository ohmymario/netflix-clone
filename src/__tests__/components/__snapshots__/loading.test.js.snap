// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading/> renders the <Loading.ReleaseBody/> 1`] = `null`;

exports[`<Loading/> renders the <Loading/> component 1`] = `
<div
  class="loading__Spinner-sc-1j6l9vs-0 bRbnAB"
  data-testid="loading"
>
  <img
    class="loading__Picture-sc-1j6l9vs-1 klMuKU"
    data-testid="loading-picture"
    src="/images/users/.png"
  />
</div>
`;
